src/components/NcRichText/NcRichText.vue

```diff
@@ -462,10 +462,10 @@ export default {
 										},
 										on: {
 											'update:checked': (value) => {
-												this.$emit('interact:todo', { id, label, value })
+												this.$emit('interact:todo', { id, label: labelParts.join(''), value })
 											},
 										},
-									}, [label])
+									}, [labelParts])
```

Comment:
As `labelParts` is an array already:

```suggestion
									}, labelParts)
```